Lib/sqlite3/__main__.py

```diff
@@ -41,41 +41,206 @@ def execute(c, sql, suppress_errors=True, theme=theme_no_color):
 
 class SqliteInteractiveConsole(InteractiveConsole):
     """A simple SQLite REPL."""
+    PS1 = "sqlite> "
+    PS2 = "    ... "
+    ruler = "="
+    doc_header = "Documented commands (type .help <command>):"
+    undoc_header = "Undocumented commands:"
 
     def __init__(self, connection, use_color=False):
         super().__init__()
         self._con = connection
         self._cur = connection.cursor()
         self._use_color = use_color
+        self._theme = get_theme(force_no_color=not use_color)
+
+        s = self._theme.syntax
+        sys.ps1 = f"{s.prompt}{self.PS1}{s.reset}"
+        sys.ps2 = f"{s.prompt}{self.PS2}{s.reset}"
+
+    def do_version(self, _):
+        """.version
+
+        Show version of the runtime SQLite library.
+        """
+        print(sqlite3.sqlite_version)
+
+    def do_help(self, arg):
```

Comment:
```suggestion
    def print_help(self, arg):
```

```suggestion
    def display_help(self, arg):
```

*do* is weird, please pick a clearer one. It currently sounds like a helper.

Reply:
The `do_` function name prefix is used to mark the dot command, like this example:

```python
class A:
    def do_help(self):
        cmds = sorted(name[3:] for name in dir(self.__class__)
                      if name.startswith("do_"))
        for cmd in cmds:
            print(cmd)
    def do_version(self): ...
    def do_quit(self): ...
    def other(self): ...

a = A()
a.do_help()

# output:
# help
# quit
# version
```

I referred this mechanism of storing command list from [cmd.Cmd](https://github.com/tanloong/cpython/blob/sqlite3-cli-help-more-helpful/Lib/cmd.py#L8), and the prefix choice `do_` is also from there.

Reply:
There is a reason why it's `do_` as well. It's what we call a visitor pattern.